mycroft/client/speech/hotword_factory.py

```diff
@@ -0,0 +1,142 @@
+                f.write(word + ' ' + phoneme + '\n')
+        return file_name
+
+    def create_config(self, dict_name, Decoder):
+        config = Decoder.default_config()
+        model_file = join(BASEDIR, 'model', self.lang, 'hmm')
+        if not exists(model_file):
+            LOG.error('PocketSphinx model not found for {}', self.lang)
+            model_file = join(BASEDIR, 'model', 'en-us', 'hmm')
+
+        config.set_string('-hmm', model_file)
+        config.set_string('-dict', dict_name)
+        config.set_string('-keyphrase', self.key_phrase)
+        config.set_float('-kws_threshold', float(self.threshold))
+        config.set_float('-samprate', self.sample_rate)
+        config.set_int('-nfft', 2048)
+        config.set_string('-logfn', '/dev/null')
+        return config
+
+    def transcribe(self, byte_data, metrics=None):
+        start = time.time()
+        self.decoder.start_utt()
+        self.decoder.process_raw(byte_data, False, False)
+        self.decoder.end_utt()
+        if metrics:
+            metrics.timer("mycroft.stt.local.time_s", time.time() - start)
+        return self.decoder.hyp()
+
+    def found_wake_word(self, frame_data):
+        hyp = self.transcribe(frame_data)
+        return hyp and self.key_phrase in hyp.hypstr.lower()
+
+
+class SnowboyHotWord(HotWordEngine):
+    def __init__(self, key_phrase, config=None, lang="en-us"):
+        super(SnowboyHotWord, self).__init__(key_phrase, config, lang)
+        # Hotword module imports
+        from snowboydecoder import HotwordDetector
+        # Hotword module config
+        module = self.config.get("module")
+        if module != "snowboy":
+            LOG.warning(module + " module does not match with Hotword class "
+                                 "snowboy")
+        # Hotword params
+        models = self.config.get("models", {})
+        paths = []
+        for key in models:
+            paths.append(models[key])
+        sensitivity = self.config.get("sensitivity", 0.5)
+        self.snowboy = HotwordDetector(paths,
+                                       sensitivity=[sensitivity] * len(paths))
+        self.lang = str(lang).lower()
+        self.key_phrase = str(key_phrase).lower()
+
+    def found_wake_word(self, frame_data):
+        wake_word = self.snowboy.detector.RunDetection(frame_data)
+        return wake_word == 1
+
+
+class HotWordFactory(object):
+    CLASSES = {
+        "pocketsphinx": PocketsphinxHotWord,
+        "snowboy": SnowboyHotWord
+    }
+
+    @staticmethod
+    def create_hotword(hotword, lang="en-us"):
+        LOG.info("creating " + hotword)
+        config = ConfigurationManager.get().get("hot_words", {})
+        module = config.get(hotword).get("module")
+        config = config.get(hotword, {"module": module})
+        clazz = HotWordFactory.CLASSES.get(module)
+        return clazz(hotword, config, lang=lang)
+
+    @staticmethod
+    def create_wake_word(lang="en-us"):
```

Comment:
Can `create_wake_word`, `create_hotword`, and `create_standup_word` be put into one function? For instance, `create_wake_word` could just be:
```
conf = ConfigurationManager.get()
return HotWordFactory.create_hotword(conf['wake_word'], conf['threshold'], conf['module'])
```

Reply:
changing